core/src/main/java/org/apache/iceberg/Partitioning.java

```diff
@@ -246,4 +246,45 @@ private static boolean equivalentIgnoringNames(PartitionField field, PartitionFi
   private static boolean compatibleTransforms(Transform<?, ?> t1, Transform<?, ?> t2) {
     return t1.equals(t2) || t1.equals(Transforms.alwaysNull()) || t2.equals(Transforms.alwaysNull());
   }
+
+  /**
+   * Adapts the provided partition data to match the table partition type built using {@link #partitionType(Table)}.
+   *
+   * @param partitionType a table partition type that includes partition fields from all specs
```

Comment:
What is the behavior for when the partition does not contain a field that is required to create the target partition type

Like if I add a column, and then partition on that column.

Reply:
You mean what if the table partition type (which is a union of all spec types) contains a field that is not in the struct for a particular spec?

Reply:
Yeah, We are just setting that to null right?

Reply:
Yeah, it will be set to null as all partition fields are optional and the table partition type is a union of all spec types.

Reply:
Resolving this as this has been discussed.